src/CabalHelper/Compiletime/Compat/Environment.hs

```diff
@@ -1,6 +1,21 @@
+{-# LANGUAGE CPP #-}
 module CabalHelper.Compiletime.Compat.Environment where
 
-import System.Environment
+import qualified System.Environment as SE (getEnvironment, setEnv)
+#ifndef mingw32_HOST_OS
+import qualified System.Posix.Env as PE (setEnv)
+#endif
 
 lookupEnv :: String -> IO (Maybe String)
-lookupEnv var = do env <- getEnvironment; return (lookup var env)
+lookupEnv var = do env <- SE.getEnvironment; return (lookup var env)
+
+setEnv :: String -> String -> Bool -> IO ()
```

Comment:
I don't like the amount of logic in here and the fact that we have to special case dependency bounds on `base`. An easier (and more backwards compatible) way to replace the use of `setEnv` would be to use [`CreateProcess`'s `env` field](https://hackage.haskell.org/package/process-1.6.1.0/docs/System-Process.html#v:env) when spawning cabal commands in the tests.

Reply:
On second thought that would need a considerable amount of refactoring since we call `cabal` deep in `installCabal`. Ugh how can something as trivial as `setEnv` not be supported prior to `base-4.5`...


In that case just get rid of the `overwrite` argument, it's not like we ever call it with `False` ;)

Also I prefer using fully qualified module names in `Compat` modules, like in https://github.com/DanielG/cabal-helper/blob/master/src/CabalHelper/Compiletime/Compat/Version.hs.